src/Facebook/InstantArticles/Transformer/Rules/GlobalRule.php

```diff
@@ -26,67 +29,69 @@ class GlobalRule extends ConfigurationSelectorRule
     const PROPERTY_TIME_PUBLISHED = 'article.publish';
     const PROPERTY_GLOBAL_BODY = 'article.body';
 
-    public function getContextClass()
+    public function getContextClass(): Vector<string>
     {
-        return InstantArticle::getClassName();
+        return Vector { InstantArticle::getClassName() };
     }
 
-    public static function create()
+    public static function create(): GlobalRule
     {
         return new GlobalRule();
     }
 
-    public static function createFrom($configuration)
+    public static function createFrom(array<string, mixed> $configuration): GlobalRule
     {
         $rule = GlobalRule::create();
 
-        $rule->withSelector($configuration['selector']);
-        $properties = $configuration['properties'];
+        $rule->withSelector(Type::mixedToString($configuration['selector']));
         $rule->withProperties(
-            [
+            Vector {
                 self::PROPERTY_GLOBAL_AUTHOR_URL,
                 self::PROPERTY_GLOBAL_AUTHOR_NAME,
                 self::PROPERTY_GLOBAL_AUTHOR_DESCRIPTION,
                 self::PROPERTY_GLOBAL_AUTHOR_ROLE_CONTRIBUTION,
                 self::PROPERTY_GLOBAL_CANONICAL_URL,
                 self::PROPERTY_GLOBAL_TITLE,
                 self::PROPERTY_TIME_PUBLISHED,
-                self::PROPERTY_GLOBAL_BODY
-            ],
-            $properties
+                self::PROPERTY_GLOBAL_BODY,
```

Comment:
Let's make sure we merge #279 into the hhvm branch as well.

Reply:
Agreed.